### DISCUSSION TITLE
Is `ignore_tags` working as expected here?

### DISCUSSION THREAD

**Discussion by bukowa** [orhun/git-cliff#707] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

I am not sure if this is working as intended. Given:
Config:
```toml
ignore_tags = ".*beta"
```
Script:
```sh
git commit --allow-empty -m "feat: add first beta feature"
git tag v1.0.0-beta.1

git commit --allow-empty -m "feat: add second beta feature"
git tag v1.0.0-beta.2

git tag v1.0.0

git commit --allow-empty -m "fix: simple fix"
git tag v1.0.1
```
Output:
```markdown
## [1.0.1] - 2021-01-23

### Feat

- Add first beta feature
- Add first beta feature
- Add second beta feature

### Fix

- Simple fix

<!-- generated by git-cliff -->
```

### Steps To Reproduce

See https://github.com/bukforks/git-cliff/tree/c9300c5fb15e5e425965dc771ab7739396ea3258

### Expected behavior

```markdown
## [1.0.1] - 2021-01-23

### Fix

- Simple fix

## [1.0.0] - 2021-01-23

### Feat

- Add first beta feature
- Add second beta feature

<!-- generated by git-cliff -->
```

### Software information

- Operating system: `Windows`
- Rust version: `rustc 1.78.0 (9b00956e5 2024-04-29)`
- Project version: `2.3.0`

**Comment by orhun** [comment]
Yes, it works as expected. Take a look at the following history:

```
* 8a0c36e (HEAD -> master, tag: v1.0.1) fix: simple fix
* 23dec65 (tag: v1.0.0-beta.2, tag: v1.0.0) feat: add second beta feature
* 1d62b05 (tag: v1.0.0-beta.1) feat: add first beta feature
```

```
 TRACE git_cliff > Ignoring release: v1.0.0-beta.2
 TRACE git_cliff > Ignoring release: v1.0.0-beta.1
```

Only tag left is `v1.0.1` so the changelog contains only that release.

`v1.0.0` is also ignored because it belongs to the same commit as `v1.0.0`. It might be an issue but I'm not sure what we can do about it.

Also see #149

**Reply by bukowa** [reply]
>v1.0.0 is also ignored because it belongs to the same commit as v1.0.0

Typo? Just making sure; probably; `v1.0.0 is also ignored because it belongs to the same commit as v1.0.0-beta.2` ?

**Reply by orhun** [reply]
Oh, yeah. I meant that 😅

**Reply by bukowa** [reply]
> Yes, it works as expected. Take a look at the following history:

I understand but still I would say it's not working as expected because my regex is not matching this tag. That makes sense?

**Reply by orhun** [reply]
Yes, but the other matched tag belongs to the same commit in this case...

**Comment by arthurgustin** [comment]
@orhun Hey, just found the link of this discussion on your website.
I have a use-case in my company where we want to implement 2 tagging strategies at the same time: semver and sprint numbered or calendar versionning.

- Each merge requests increment the semver
- At the end of each sprint we tag the latest version with the number of the sprint, so one commit has 2 tags (2.2.0 and SPRINT.3 for instance)

I can not find a set configuration in git-cliff that would allow this behavior... Is it possible and if not can you consider adding this feature ?

**Reply by arthurgustin** [reply]
Just answered my own question 3 minutes later... By using this regex `tag_pattern = '[0-9]+\.[0-9]+\.[0-9]+$'` it seems to increment correctly the semver even with another tag format in the history

**Reply by orhun** [reply]
Ah cool, let me know if you need any further help!
